lib/reverse_proxy_plug/http_client/adapters/tesla.ex

```diff
@@ -0,0 +1,47 @@
+defmodule ReverseProxyPlug.HTTPClient.Adapters.Tesla do
+  @moduledoc """
+  Tesla adapter for the `ReverseProxyPlug.HTTPClient` behaviour
+
+  Only synchronous responses are supported.
+
+  ## Options
+
+  * `:tesla_client` - mandatory definition for the `Tesla.Client`
+                      to be used.
+  """
+
+  alias ReverseProxyPlug.HTTPClient
+
+  @behaviour HTTPClient
+
+  @impl HTTPClient
+  def request(%HTTPClient.Request{options: options} = request) do
+    {client, opts} = Keyword.pop(options, :tesla_client)
```

Comment:
Suggestions are welcome on this option. It is the `Tesla.Client` which is to be used with the request